database/models/user_models.py

```diff
@@ -34,7 +35,7 @@ class Teacher(models.Model):
     The teacher / instructor data model.
     """
     user = models.ForeignKey("accounts.BaseUser", on_delete=models.CASCADE, blank=False, null=False)
-    department = models.ForeignKey("database.Department", on_delete=models.CASCADE, null=True, blank=True)
+    department = models.ForeignKey("database.Department", on_delete=models.CASCADE, null=True, blank=False)
```

Comment:
blank should be True so that you can add teachers without a specific department

Reply:
noted